hardware-memory-isolation/README.md

```diff
@@ -0,0 +1,221 @@
+# Hardware Memory Isolation
+
+## Table of Contents
+
+## Prerequisites
+
+## Introduction
+
+Another role of the hardware, besides computing, is to provide isolation, mainly between different software components.
```

Comment:
If you choose to go with this statement, after @teodutu's comments, please rephrase it, just a bit.
```suggestion
Besides computing, another role of the hardware, is to provide isolation, mainly between different software components.
```